drivers/clock_control/clock_control_wch_rcc.c

```diff
@@ -113,6 +117,7 @@ static int clock_control_wch_rcc_init(const struct device *dev)
 		}
 	}
 
+#if defined(CONFIG_SOC_SERIES_CH32V00X)
```

Comment:
We should ask somebody else to confirm, but I don't believe the `#if defined SOC...` is the way to handle differences between the peripherals

Instead, this should work off the Devicetree properties directly via `IS_ENABLED` and similar.

Reply:
I think so too, ill change it to use the DT properties instead of the kconfig define.

Reply:
Im thinking it is possible here to replace the IS_ENABLED C if with a preprocessor if defined with the same config variable and maybe save a couple bytes.